self-tests fail against aiohttp 3.12.14

With upgrade `aiohttp` `3.12.13` => `3.12.14` several tests fail:
```
======================================================= short test summary info ========================================================
FAILED tests/test_ssdp.py::test_decode_ssdp_packet - aiohttp.http_exceptions.InvalidHeader: 400, message:
FAILED tests/test_ssdp.py::test_decode_ssdp_packet_missing_ending - aiohttp.http_exceptions.InvalidHeader: 400, message:
FAILED tests/test_ssdp.py::test_decode_ssdp_packet_duplicate_header - aiohttp.http_exceptions.InvalidHeader: 400, message:
FAILED tests/test_ssdp.py::test_decode_ssdp_packet_empty_location - aiohttp.http_exceptions.InvalidHeader: 400, message:
FAILED tests/test_ssdp.py::test_decode_ssdp_packet_v6 - aiohttp.http_exceptions.InvalidHeader: 400, message:
FAILED tests/test_ssdp.py::test_microsoft_butchers_ssdp - aiohttp.http_exceptions.InvalidHeader: 400, message:
```

with similar error:
```
_______________________________________________________ test_decode_ssdp_packet ________________________________________________________

    def test_decode_ssdp_packet() -> None:
        """Test SSDP response decoding."""
        msg = (
            b"HTTP/1.1 200 OK\r\n"
            b"Cache-Control: max-age=1900\r\n"
            b"Location: http://192.168.1.1:80/RootDevice.xml\r\n"
            b"Server: UPnP/1.0 UPnP/1.0 UPnP-Device-Host/1.0\r\n"
            b"ST:urn:schemas-upnp-org:service:WANCommonInterfaceConfig:1\r\n"
            b"USN: uuid:...::WANCommonInterfaceConfig:1\r\n"
            b"EXT:\r\n\r\n"
        )
>       request_line, headers = decode_ssdp_packet(
            msg, ("local_addr", 1900), ("remote_addr", 12345)
        )

tests/test_ssdp.py:74:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
async_upnp_client/ssdp.py:251: in decode_ssdp_packet
    request_line, headers = _cached_decode_ssdp_packet(data, remote_addr_without_port)
async_upnp_client/ssdp.py:216: in _cached_decode_ssdp_packet
    parsed_headers, request_line, udn = _cached_header_parse(data)
async_upnp_client/ssdp.py:192: in _cached_header_parse
    parsed_headers, _ = HeadersParser().parse_headers(lines)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <aiohttp.http_parser.HeadersParser object at 0x7f23520f3770>
lines = [b'HTTP/1.1 200 OK', b'Cache-Control: max-age=1900', b'Location: http://192.168.1.1:80/RootDevice.xml', b'Server: UPnP...1.0', b'ST:urn:schemas-upnp-org:service:WANCommonInterfaceConfig:1', b'USN: uuid:...::WANCommonInterfaceConfig:1', ...]

    def parse_headers(
        self, lines: List[bytes]
    ) -> Tuple["CIMultiDictProxy[str]", RawHeaders]:
        headers: CIMultiDict[str] = CIMultiDict()
        # note: "raw" does not mean inclusion of OWS before/after the field value
        raw_headers = []
    
        lines_idx = 0
        line = lines[lines_idx]
        line_count = len(lines)
    
        while line:
            # Parse initial header name : value pair.
            try:
                bname, bvalue = line.split(b":", 1)
            except ValueError:
>               raise InvalidHeader(line) from None
E               aiohttp.http_exceptions.InvalidHeader: 400, message:
E                 Invalid HTTP header: b'HTTP/1.1 200 OK'

.tox/py3/lib64/python3/site-packages/aiohttp/http_parser.py:154: InvalidHeader
```

--- a/.pre-commit-config.yaml
+++ b/.pre-commit-config.yaml
@@ -64,7 +64,7 @@ repos:
         additional_dependencies:
           - python-didl-lite~=1.4.0
           - pytest~=8.3.3
-          - aiohttp~=3.10.9
+          - aiohttp~=3.12.14
           - pytest-asyncio >= 0.24,< 0.27
           - pytest-aiohttp >=1.0.5,<1.2.0
         files: ^(async_upnp_client|tests)/.+\.py$

--- a/async_upnp_client/profiles/dlna.py
+++ b/async_upnp_client/profiles/dlna.py
@@ -84,7 +84,7 @@ class DlnaOrgCi(Enum):
 
 
 class DlnaOrgPs(Enum):
-    """DLNA.ORG_PS (PlaySpeed ) flags."""
+    """DLNA.ORG_PS (PlaySpeed) flags."""
 
     INVALID = 0
     NORMAL = 1

--- a/async_upnp_client/ssdp.py
+++ b/async_upnp_client/ssdp.py
@@ -183,13 +183,13 @@ def _cached_header_parse(
     """
     lines = data.replace(b"\r\n", b"\n").split(b"\n")
 
-    # request_line
     request_line = lines[0].strip().decode()
 
-    if lines and lines[-1] != b"":
-        lines.append(b"")
+    header_lines = lines[1:] if lines else []
+    if header_lines and header_lines[-1] != b"":
+        header_lines.append(b"")
 
-    parsed_headers, _ = HeadersParser().parse_headers(lines)
+    parsed_headers, _ = HeadersParser().parse_headers(header_lines)
 
     usn = parsed_headers.get("usn")
     udn = udn_from_usn(usn) if usn else None

--- a/changes/273.bugfix
+++ b/changes/273.bugfix
@@ -1,0 +1,3 @@
+Fix SSDP header parsing.
+
+aiohttp 3.12.14 starts failing due to including the request line in the header lines to be parsed.

--- a/tests/test_server.py
+++ b/tests/test_server.py
@@ -190,7 +190,7 @@ async def upnp_server(monkeypatch: Any, aiohttp_client: Any) -> AsyncGenerator:
     """Fixture to initialize device."""
     # pylint: disable=too-few-public-methods
 
-    ssdp_sockets = []
+    ssdp_sockets: List[socket.socket] = []
     http_client = None
 
     def get_ssdp_socket_mock(
@@ -231,6 +231,7 @@ async def upnp_server(monkeypatch: Any, aiohttp_client: Any) -> AsyncGenerator:
 
     assert aiohttp_client
     await callback.start(aiohttp_client)
+    assert http_client
     yield UpnpServerTuple(http_client, ssdp_sockets, callback, server)
     # await server.async_stop()
     for sock in ssdp_sockets:

--- a/tests/test_server.py
+++ b/tests/test_server.py
@@ -22,10 +22,11 @@ from unittest.mock import Mock
 import aiohttp
 import pytest
 import pytest_asyncio
+from aiohttp.test_utils import TestClient
+from aiohttp.web import Application, Request
 from pytest_aiohttp.plugin import AiohttpClient
 
 import async_upnp_client.aiohttp
-import async_upnp_client.client
 import async_upnp_client.server
 from async_upnp_client.client import UpnpStateVariable
 from async_upnp_client.const import DeviceInfo, ServiceInfo
@@ -149,11 +150,11 @@ class Callback:
         self.callback: Optional[
             Callable[[aiohttp.web.Request], Awaitable[aiohttp.web.Response]]
         ] = None
-        self.session = None
+        self.session: TestClient[Request, Application] | None = None
         self.app = aiohttp.web.Application()
         self.app.router.add_route("NOTIFY", "/{tail:.*}", self.handler)
 
-    async def start(self, aiohttp_client: Any) -> None:
+    async def start(self, aiohttp_client: AiohttpClient) -> None:
         """Generate session."""
         self.session = await aiohttp_client(self.app)
 
@@ -179,14 +180,16 @@ class Callback:
 class UpnpServerTuple(NamedTuple):
     """Upnp server tuple."""
 
-    http_client: AiohttpClient
+    http_client: TestClient
     ssdp_sockets: List[socket.socket]
     callback: Callback
     server: UpnpServer
 
 
 @pytest_asyncio.fixture
-async def upnp_server(monkeypatch: Any, aiohttp_client: Any) -> AsyncGenerator:
+async def upnp_server(
+    monkeypatch: Any, aiohttp_client: AiohttpClient
+) -> AsyncGenerator[UpnpServerTuple, None]:
     """Fixture to initialize device."""
     # pylint: disable=too-few-public-methods
 
@@ -212,7 +215,7 @@ async def upnp_server(monkeypatch: Any, aiohttp_client: Any) -> AsyncGenerator:
         async def start(self) -> Any:
             """Create HTTP server."""
             nonlocal http_client
-            http_client = cast(AiohttpClient, await aiohttp_client(self.app))
+            http_client = cast(TestClient, await aiohttp_client(self.app))
             return http_client
 
     callback = Callback()
@@ -229,9 +232,9 @@ async def upnp_server(monkeypatch: Any, aiohttp_client: Any) -> AsyncGenerator:
     )
     await server.async_start()
 
+    assert http_client
     assert aiohttp_client
     await callback.start(aiohttp_client)
-    assert http_client
     yield UpnpServerTuple(http_client, ssdp_sockets, callback, server)
     # await server.async_stop()
     for sock in ssdp_sockets:

--- a/tests/test_server.py
+++ b/tests/test_server.py
@@ -215,7 +215,7 @@ async def upnp_server(
         async def start(self) -> Any:
             """Create HTTP server."""
             nonlocal http_client
-            http_client = cast(TestClient, await aiohttp_client(self.app))
+            http_client = await aiohttp_client(self.app)
             return http_client
 
     callback = Callback()

--- a/tests/test_ssdp.py
+++ b/tests/test_ssdp.py
@@ -55,7 +55,8 @@ def test_is_valid_ssdp_packet() -> None:
         b"Server: UPnP/1.0 UPnP/1.0 UPnP-Device-Host/1.0\r\n"
         b"ST:urn:schemas-upnp-org:service:WANCommonInterfaceConfig:1\r\n"
         b"USN: uuid:...::WANCommonInterfaceConfig:1\r\n"
-        b"EXT:\r\n\r\n"
+        b"EXT:\r\n"
+        b"\r\n"
     )
     assert is_valid_ssdp_packet(msg)
 
@@ -69,7 +70,8 @@ def test_decode_ssdp_packet() -> None:
         b"Server: UPnP/1.0 UPnP/1.0 UPnP-Device-Host/1.0\r\n"
         b"ST:urn:schemas-upnp-org:service:WANCommonInterfaceConfig:1\r\n"
         b"USN: uuid:...::WANCommonInterfaceConfig:1\r\n"
-        b"EXT:\r\n\r\n"
+        b"EXT:\r\n"
+        b"\r\n"
     )
     request_line, headers = decode_ssdp_packet(
         msg, ("local_addr", 1900), ("remote_addr", 12345)
@@ -137,7 +139,8 @@ def test_decode_ssdp_packet_duplicate_header() -> None:
     msg = (
         b"HTTP/1.1 200 OK\r\n"
         b"CACHE-CONTROL: max-age = 1800\r\n"
-        b"CACHE-CONTROL: max-age = 1800\r\n\r\n"
+        b"CACHE-CONTROL: max-age = 1800\r\n"
+        b"\r\n"
     )
     _, headers = decode_ssdp_packet(msg, ("local_addr", 1900), ("remote_addr", 12345))
 
@@ -156,7 +159,8 @@ def test_decode_ssdp_packet_empty_location() -> None:
     msg = (
         b"HTTP/1.1 200 OK\r\n"
         b"LOCATION: \r\n"
-        b"CACHE-CONTROL: max-age = 1800\r\n\r\n"
+        b"CACHE-CONTROL: max-age = 1800\r\n"
+        b"\r\n"
     )
     _, headers = decode_ssdp_packet(msg, ("local_addr", 1900), ("remote_addr", 12345))
 
@@ -174,7 +178,12 @@ def test_decode_ssdp_packet_empty_location() -> None:
 @pytest.mark.asyncio
 async def test_ssdp_protocol_handles_broken_headers() -> None:
     """Test SsdpProtocol is able to handle broken headers."""
-    msg = b"HTTP/1.1 200 OK\r\n" b"DEFUNCT\r\n" b"CACHE-CONTROL: max-age = 1800\r\n\r\n"
+    msg = (
+        b"HTTP/1.1 200 OK\r\n"
+        b"DEFUNCT\r\n"
+        b"CACHE-CONTROL: max-age = 1800\r\n"
+        b"\r\n"
+    )
     addr = ("addr", 123)
     loop = asyncio.get_event_loop()
 
@@ -194,7 +203,8 @@ def test_decode_ssdp_packet_v6() -> None:
         b"Server: UPnP/1.0 UPnP/1.0 UPnP-Device-Host/1.0\r\n"
         b"ST:urn:schemas-upnp-org:service:WANCommonInterfaceConfig:1\r\n"
         b"USN: uuid:...::WANCommonInterfaceConfig:1\r\n"
-        b"EXT:\r\n\r\n"
+        b"EXT:\r\n"
+        b"\r\n"
     )
 
     request_line, headers = decode_ssdp_packet(

--- a/tox.ini
+++ b/tox.ini
@@ -14,7 +14,7 @@ commands = py.test --cov=async_upnp_client --cov-report=term --cov-report=xml:co
 ignore_errors = True
 deps =
     pytest == 8.3.3
-    aiohttp ~= 3.10.9
+    aiohttp ~= 3.12.14
     pytest-asyncio >= 0.24,< 0.27
     pytest-aiohttp >=1.0.5,<1.2.0
     pytest-cov >= 5.0,< 6.2
@@ -54,7 +54,7 @@ deps =
     mypy == 1.11.2
     python-didl-lite ~= 1.4.0
     pytest ~= 8.3.3
-    aiohttp ~= 3.10.9
+    aiohttp ~= 3.12.14
     pytest-asyncio >= 0.24,< 0.27
     pytest-aiohttp >=1.0.5,<1.2.0
 commands = mypy --ignore-missing-imports async_upnp_client tests

